hazelcast-jet-core/src/main/java/com/hazelcast/jet/core/processor/Processors.java

```diff
@@ -403,6 +427,16 @@ private Processors() {
      * it deletes from storage all the frames that trail behind the emitted
      * windows. The type of emitted items is {@link TimestampedEntry
      * TimestampedEntry&lt;K, A>} so there is one item per key per window position.
+     * <p>
+     * <i>Behavior on job restart</i><br>
+     * This processor saves its state to snapshot. After restart, it can
+     * continue accumulating where it left off.
+     * <p>
+     * In at-least-once mode, the watermarks are allowed to resume at lower
```

Comment:
I think this sentence is a bit hard to understand: My attempt:

"In at-least-once mode, after a restart the emitted windows can contain duplicate events that were already processed before the last snapshot. Furthermore, at-least-once allows watermarks to go back in time after a restart. If such a watermark is received, some windows that were emitted before the snapshot will be re-emitted. These windows might contain less events than before as some of them had already been evicted before the snapshot."